e2e-webdriver/src/page-objects/app.po.ts

```diff
@@ -0,0 +1,18 @@
+export class AppPage {
+
+  async navigateTo(): Promise<string> {
+    return browser.url('');
+  }
+
+  async getTitle(): Promise<string> {
+    const $h1: WebdriverIO.Element = await $('app-root h1');
+    return $h1.getText();
+  }
+
+  async getProductLinkByName(name: string): Promise<WebdriverIO.Element> {
```

Comment:
Für die Produktübersichtsseite (app-product-master) würde ich ein dediziertes Page Object machen. Die `AppPage` hingegen sollte nur allgemeine, überall sichtbare Dinge enthalten. So wie eben der Titel (h1).

Reply:
Bin mir nicht sicher ob ich da mitgehe.
`get title()` ist spezifisch pro Seite (in der master-page ist der Titel im `h1`, in der detail-page im `h2`), `navigateTo()` und `getProductImageByName(name: string)` auch. D.h. ich finde in app.po.ts nichts, was in ein allgemeines Page Object ausgelagert werden könnte.
Vielleicht könnte man aber app.po.ts in product-master.po.ts umbenennen, damit es klarer wird.